File: src/service/scrappingService.ts
import { CheerioWebBaseLoader } from "@langchain/community/document_loaders/web/cheerio";

class ScrappingService {
  constructor() {}

  async getWebPageContent(url: string) {
    const loader = new CheerioWebBaseLoader(url);
    const docs = await loader.load();
    return docs;
  }
}

const scrappingService = new ScrappingService();
export default scrappingService;

File: fastapi/service/Scrappingservice.py
from typing import List
from langchain_core.documents import Document
from langchain_community.document_loaders import WebBaseLoader


class Scrappingservice(object):
    """docstring for Scrappingservice."""

    def __init__(self):
        super(Scrappingservice, self).__init__()

    def getWebPageContent(self, url: str) -> List[Document]:
        loader = WebBaseLoader(url)
        data = loader.load()
        return data


scrappingservice = Scrappingservice()
